akka-bench-jmh/src/main/scala/akka/stream/MaterializationBenchmark.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.stream
+
+import java.util.concurrent.TimeUnit
+import akka.actor.ActorSystem
+import akka.stream.scaladsl._
+import org.openjdk.jmh.annotations._
+
+object MaterializationBenchmark {
+
+  val complexGraphBuilder = () =>
+    FlowGraph.closed() { implicit b =>
+      val numOfJunctions = 100
+
+      import FlowGraph.Implicits._
+      val broadcast = b.add(Broadcast[Unit](numOfJunctions))
+      var outlet = broadcast.out(0)
+      for (i <- 1 until numOfJunctions) {
+        val merge = b.add(Merge[Unit](2))
+        outlet ~> merge
+        broadcast.out(i) ~> merge
+        outlet = merge.out
+      }
+
+      Source.single(()) ~> broadcast
+      outlet ~> Sink.ignore
+    }
+
+  val graphWithNestedImportsBuilder = () => {
+    var flow: Graph[FlowShape[Unit, Unit], Unit] = Flow[Unit].map(identity)
+    for (_ <- 1 to 100) {
+      flow = FlowGraph.partial(flow) { implicit b ⇒
```

Comment:
what does this intend to do? Btw, I would prefer calling b.add() instead. 
What I would like to see is a graph that is imported 1000 times to another graph. Then we can compare that with creating 1000 graphs directly instead of importing.